msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -499,6 +532,25 @@ public List getAllUserStatistics(){
 		//clear any gradebook info:
 		resetGradebookVariables();
 
+		// Determine whether the site is a pure anonymous scenario (Ie. all topics in the site are anonymous)
+		String siteId = ToolManager.getCurrentPlacement().getContext();
+		List<Topic> allTopics = forumManager.getTopicsInSite(siteId);
```

Comment:
Wouldn't work in situations where every topic in the site has POST_ANONYMOUS = true, and REVEAL_IDS_TO_ROLES = true, and the user has the identifyAnonAuthors permission.

Though you bring up a good point because most sites will have the "General Discussion" topic which should be revealed 99.9% of the time. So in the common case we avoid this loop and use quick query like:
select count(*) from MFR_TOPIC_T topic (join ... to tie it to the context) where topic.post_anonymous=1 limit 1;